tardis/plugins/auditor.py

```diff
@@ -85,7 +89,21 @@ async def notify(self, state: State, resource_attributes: AttributeDict) -> None
                 .replace(tzinfo=self._local_timezone)
                 .astimezone(pytz.utc)
             )
-            await self._client.update(record)
+            try:
+                await self._client.update(record)
+            except RuntimeError as e:
+                reqwest_error_match = REQWEST_ERROR.match(str(e))
+                if reqwest_error_match:
+                    http_error_code = reqwest_error_match.group(1)
+                    if http_error_code == "400":
```

Comment:
It looks like these double-`if` could be folded into one, making it more obvious that you are really just checking *one* thing and unconditionally ``raise`` otherwise.

Either...

Reply:
match `and` check the code:
```suggestion
                error_match = REQWEST_ERROR.match(str(e))
                if error_match and error_match.group(1) == "400:
                    ...
```
(this could also do with an assignment expression, but is rather dense then)
or ...

Reply:
make a literal match against the error, since effectively this is what you are doing:
```suggestion
                if str(e).startswith("Reqwest Error: HTTP status client error (400 Bad Request)"):
```
In this case I would recommend to create a new constant for the error message instead of the pattern. Note that I've guessed the necessity for `.startswith` from the pattern's trailing `.*`, it may be unnecessary or not.

Reply:
@maxfischer2781 thanks and 👍 for your remarks. Just a reminder that assignment expressions would not work until #304 is approved and merged.

Reply:
Thanks for the suggestion. I agree that maybe the error message matching was slightly overengineered for this case :wink:  I've replaced the regex logic with a simple `str(e).startswith(...)`.